lib/internal/Magento/Framework/File/Name.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Framework\File;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Validation\ValidationException;
+
+/**
+ * Class Name
+ *
+ * @package Magento\Framework\File
+ */
+class Name
+{
+    /**
+     * Get new file name if the given name is in use
+     *
+     * @param string $destinationFile
+     * @return string
+     */
+    public function getNewFileName(string $destinationFile)
```

Comment:
Please, declare `strict_types` for this class and use the return types hinting for class methods